### PR TITLE
Instructor: students: all records: fix error when course has a session with null startTime #8547

### LINKED ISSUE #8547: Instructor: students: all records: fix error when course has a session with null startTime
**Environment**
`master` branch at 3670601713a2a0211067334096a8a99f73fb3da0

**Steps to reproduce:**
- Login as `Instructor`
- Go to `Students` and expand a `Course` too see all the students in the course
- Click on any of the students `All Records` option (This is to check that we can access the student records)
You will get something like this : 
screenshot 389

Now, 
- Go to `Sessions`
- Create a `private` session without specifying a `Submission Opening Time`
- Go back to `Students` page, and expand the course where you created the `Private` session
- Click on any of the students `All Records` option. You will get this error.
screenshot 386

**Expected behaviour:** This error should not occur. 

**Actual behaviour:** Error occurs.

**P.S.:** This bug will get fixed automatically after #8447 is merged

### PR DESCRIPTION
**Pull request by sukanta-27**
Fixes #8547 

**Outline of Solution**

- Fixed the issues for `DESCENDING_ORDER` comparator in session attributes (The lines that needed to be changed are shown in https://github.com/TEAMMATES/teammates/issues/8547#issuecomment-368660793 )
- Implemented the same in `ASCENDING_ORDER` too (though it is never used except in `SessionAttributesTest`)

### CODE REVIEW

**Comment by sukanta-27** [comment]
@whipermr5 A reminder.

**Comment by sukanta-27** [comment]
@whipermr5 A reminder

**Comment by sukanta-27** [comment]
@whipermr5 A reminder. :sweat_smile:

File: src/main/java/teammates/common/datatransfer/attributes/SessionAttributes.java [hunk 1]
```diff
@@ -19,20 +19,41 @@
      */
     Comparator<SessionAttributes> ASCENDING_ORDER = (session1, session2) -> {
 
+        FeedbackSessionAttributes feedbackSession1 = null;
+        FeedbackSessionAttributes feedbackSession2 = null;
+        Boolean isSession1PrivateFeedback = false;
+        Boolean isSession2PrivateFeedback = false;
+        if (session1 instanceof FeedbackSessionAttributes) {
+            feedbackSession1 = (FeedbackSessionAttributes) session1;
+        }
```
Comment (whipermr5) [inline src/main/java/teammates/common/datatransfer/attributes/SessionAttributes.java]: Understand that you're following the example of `DESCENDING_ORDER`, but a better way would be to add `isPrivateSession()` to the interface for much more concise code. Can update `DESCENDING_ORDER` too.

File: src/main/java/teammates/common/datatransfer/attributes/SessionAttributes.java [hunk 2]
```diff
@@ -19,20 +19,41 @@
...
+        if (session2 instanceof FeedbackSessionAttributes) {
+            feedbackSession2 = (FeedbackSessionAttributes) session2;
+        }
+
         Assumption.assertNotNull(session1.getSessionName());
-        Assumption.assertNotNull(session1.getSessionStartTime());
```
Comment (whipermr5) [inline src/main/java/teammates/common/datatransfer/attributes/SessionAttributes.java]: Actually, just add a guard clause at the very start. If the session is private, simply compare session name and return. Do the same for `DESCENDING_ORDER`.

**Comment by sukanta-27** [comment]
@whipermr5 Ready for review.

**Pull request merged**